lib/mllp/client.ex

```diff
@@ -407,7 +411,11 @@ defmodule MLLP.Client do
               state
             )
 
-            new_state = maintain_reconnect_timer(state)
+            new_state =
```

Comment:
If we encounter an error while trying to recv it is most likely unrecoverable and thus the best course of action is hang up the phone to clear state between the client and the server. But we also need to provide users a way to opt out of this behavior (`close_on_recv_error`).